packages/cashier/src/pages/p2p-cashier/p2p-cashier.tsx

```diff
@@ -109,9 +114,41 @@ const P2PCashier = ({
                 setOrderId(input_order_id);
             }
         },
-        [history, location.hash, location.search, order_id]
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+        [history, location.hash, location.search]
     );
 
+    React.useEffect(() => {
+        const url_params = new URLSearchParams(location.search);
+        let passed_order_id;
+
+        setActionParam(url_params.get('action'));
+        if (is_mobile) {
+            setCodeParam(localStorage.getItem('verification_code.p2p_order_confirm'));
+        } else if (!code_param) {
+            if (url_params.has('code')) {
+                setCodeParam(url_params.get('code'));
+            } else if (localStorage.getItem('verification_code.p2p_order_confirm')) {
+                setCodeParam(localStorage.getItem('verification_code.p2p_order_confirm'));
+            }
+        }
+
+        // Different emails give us different params (order / order_id),
+        // don't remove order_id since it's consistent for mobile and web for 2FA
+        if (url_params.has('order_id')) {
+            passed_order_id = url_params.get('order_id');
+        } else if (url_params.has('order')) {
+            passed_order_id = url_params.get('order');
+        }
+
+        if (passed_order_id) {
+            setQueryOrder(passed_order_id);
+        }
+
+        return () => setQueryOrder(null);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
@hamid-deriv Can’t fix it instead of ignoring it? 🤔

Reply:
I need to remove the `order_id` dependency because of [this](https://github.com/binary-com/deriv-app/pull/6519#discussion_r977281973). Also, it needs to be checked in the `setQueryOrder`, so I need to add this eslint ignore line.